samples/wavlib/src/main/cpp/io/wav/WavStreamReader.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <algorithm>
+#include <string.h>
+
+#include <android/log.h>
+
+#include "io/stream/InputStream.h"
+
+#include "AudioEncoding.h"
+#include "WavRIFFChunkHeader.h"
+#include "WavFmtChunkHeader.h"
+#include "WavChunkHeader.h"
+#include "WavStreamReader.h"
+
+static const char *TAG = "WavStreamReader";
+
+namespace wavlib {
+
+WavStreamReader::WavStreamReader(InputStream *stream) {
+    mStream = stream;
+
+    mWavChunk = 0;
+    mFmtChunk = 0;
+    mDataChunk = 0;
+
+    mAudioDataStartPos = -1;
+
+    mChunkMap = new std::map<RiffID, WavChunkHeader *>();
+}
+
+int WavStreamReader::getSampleFormat() {
+    if (mFmtChunk->mFormatId == WavFmtChunkHeader::ENCODING_PCM) {
+        return mFmtChunk->mSampleSize == 8
+               ? AudioEncoding::PCM_8
+               : AudioEncoding::PCM_16;

```

Comment:
What about 24-bit data? I think it is OK to reject if seen as UNIMPLEMENTED for now.